server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -130,12 +130,15 @@ protected void doClose() {
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
         try {
-            SnapshotsInProgress currentSnapshots = SnapshotsInProgress.get(event.state());
+            final var currentSnapshots = SnapshotsInProgress.get(event.state());
             if (SnapshotsInProgress.get(event.previousState()).equals(currentSnapshots) == false) {
+                final var localNodeId = clusterService.localNode().getId();
                 synchronized (shardSnapshots) {
                     cancelRemoved(currentSnapshots);
-                    for (List<SnapshotsInProgress.Entry> snapshots : currentSnapshots.entriesByRepo()) {
-                        startNewSnapshots(snapshots);
+                    for (final var oneRepoSnapshots : currentSnapshots.entriesByRepo()) {
+                        for (final var snapshotInProgress : oneRepoSnapshots) {
+                            handleUpdatedSnapshotInProgress(localNodeId, snapshotInProgress);
```

Comment:
Nit: Can we have a suffix of `Entry` to the variable and method names, e.g. `snapshotInProgressEntry` (or maybe just `entry`) and `handleUpdatedSnapshotInProgressEntry`. It would be great if we could have consistent variable names corresponding to classes like Snapshot, SnapshotId, SnapshotsInProgress, SnapshotsInProgressEntry etc.

Reply:
An `Entry` in the `SnapshotsInProgress` collection could reasonably be called a `SnapshotInProgress` IMO, but ++ for consistency for now.